### ISSUE TITLE
Bug  panic: runtime error: index out of range [0] with length 0

### ISSUE THREAD

**Issue by worker24h** [VictoriaMetrics/VictoriaMetrics#7762]
### Describe the bug

When I executed `_time: 5m MarsUploadServiceImpl | stream_context before 4`, then the UI of vm-logs had response `No logs found` and the backend service `victoria-logs-prod` throwed a panic.

Related PR： https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7637

### To Reproduce

Key Query:  `_time: 5m MarsUploadServiceImpl | stream_context before 4`, As follow：
![image](https://github.com/user-attachments/assets/d6365560-3557-42d8-837b-d170d316ffb6)

### Version
**The latest version v1.3.1**

/ # ./victoria-logs-prod --version
victoria-logs-20241208-011727-tags-v1.3.1-victorialogs-0-g48f395456e

### Logs

```bash
panic: runtime error: index out of range [0] with length 0

goroutine 7205 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:290 +0x5b
panic({0xa3f4c0?, 0xc011b10d80?})
	runtime/panic.go:785 +0x132
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*pipeStreamContextProcessor).getTimeRangesForStreamRowss(0xc00050e750, {0xc054cbebd0, 0x30}, {0xc00ed0c000, 0x39ed9, 0x39ed9}, 0x19f38defa)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/pipe_stream_context.go:177 +0x327
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*pipeStreamContextProcessor).getStreamRowss(0xc00050e750, {0xc054cbebd0, 0x30}, {0xc057700000, 0x39ed9, 0xc000828128?}, 0x19f55d5c2)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/pipe_stream_context.go:139 +0x13d
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*pipeStreamContextProcessor).flush(0xc00050e750)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/pipe_stream_context.go:655 +0x374
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).runQuery(0xc000220210, {0xc5e348, 0xc0026561e0}, {0xc0008280d0, 0x1, 0x1}, 0xa38140?, 0xc000ada560)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage_search.go:177 +0x9ad
github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage.(*Storage).RunQuery(0xc000220210, {0xc5e348, 0xc0026561e0}, {0xc0008280d0, 0x1, 0x1}, 0xc00038b980, 0xc00038baa0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logstorage/storage_search.go:109 +0xab
github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage.RunQuery(...)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlstorage/main.go:138
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql.getQueryResultsWithLimit({0xc5e380?, 0xc000226e70?}, {0xc0008280d0, 0x1, 0x1}, 0xc00038b980, 0x7d0)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql/logsql.go:1032 +0x185
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql.getLastNQueryResults({0xc5e380, 0xc000226e70}, {0xc0008280d0, 0x1, 0x1}, 0xc00038b980, 0x3e8)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql/logsql.go:925 +0x134
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql.ProcessQueryRequest({0xc5e380, 0xc000226e70}, {0xc5d418, 0xc0001826a8}, 0xc0004c8280)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/logsql/logsql.go:875 +0x62c
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect.processSelectRequest({0xc5e380, 0xc000226e70}, {0xc5d418, 0xc0001826a8}, 0xc0004c8280, {0xc000a5c2a5, 0x14})
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/main.go:198 +0x118
github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect.RequestHandler({0xc5d418, 0xc0001826a8}, 0xc0004c8280)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vlselect/main.go:149 +0x56e
main.requestHandler({0xc5d418, 0xc0001826a8}, 0xc0004c8280)
	github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-logs/main.go:92 +0x1a5
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc0000305a0, {0xc5dc58, 0xc00037fd60}, 0xc0004c8280, 0xbc0d70)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:415 +0xf4f
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xc5dc58?, 0xc00037fd60?}, 0x0?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:249 +0x31
net/http.HandlerFunc.ServeHTTP(0x118da90?, {0xc5dc58?, 0xc00037fd60?}, 0x4?)
	net/http/server.go:2220 +0x29
github.com/klauspost/compress/gzhttp.NewWrapper.func1.1({0xc5d328, 0xc00057ae00}, 0xc0004c8280)
	github.com/klauspost/compress@v1.17.11/gzhttp/compress.go:517 +0x5ef
net/http.HandlerFunc.ServeHTTP(0x40d8e5?, {0xc5d328?, 0xc00057ae00?}, 0xc00057ae01?)
	net/http/server.go:2220 +0x29
net/http.serverHandler.ServeHTTP({0xc5b480?}, {0xc5d328?, 0xc00057ae00?}, 0x6?)
	net/http/server.go:3210 +0x8e
net/http.(*conn).serve(0xc00050e240, {0xc5e310, 0xc00038b4a0})
	net/http/server.go:2092 +0x5d0
created by net/http.(*Server).Serve in goroutine 1179
	net/http/server.go:3360 +0x485
[root@k3s06 victorialogs]#
[root@k3s06 victorialogs]#
[root@k3s06 victorialogs]#
```

@valyala

**Comment by worker24h** [comment]
@valyala  Hi, I have read the context of vmlogs，but I’m not sure why the `rows` is empty.  I am unable to fix it at the moment.

**Comment by valyala** [comment]
@worker24h , thanks for the bugreport! The surrounding `rows` for some original log entry can be empty if they were already included in the surrounding logs of the previous original log entry. For example, if there are multiple adjacent logs with the `MarsUploadServiceImpl` word or if there are less than 4 other logs between logs with the `MarsUploadServiceImpl` word, then `MarsUploadServiceImpl | stream_context 4` would include these logs into the surrounding `rows` for the first seen original log entry with the `MarsUploadServiceImpl` word, while surrounding `rows` for subsequent logs will be empty.

This issue must be fixed in the commit 843fae341978a526e9603759fec11fd6b301982b , which will be included in the next release.

**Comment by valyala** [comment]
The bugfix has been included in the release [v1.3.2-victorialogs](https://docs.victoriametrics.com/victorialogs/changelog/). Closing the issue as fixed.
